src/libmongoc/CMakeLists.txt

```
@@ -707,20 +707,19 @@ set (STATIC_LIBRARIES
    ${SNAPPY_LIBRARIES} ${ZLIB_LIBRARIES} ${MONGOC_ZSTD_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT} ${ICU_LIBRARIES} ${LIBMONGOCRYPT_LIBRARY}
 )
 
+add_library(_mongoc-dependencies INTERFACE)
+add_library(mongo::detail::c_dependencies ALIAS _mongoc-dependencies)
+install(TARGETS _mongoc-dependencies EXPORT mongoc-targets)
+set_property(TARGET _mongoc-dependencies PROPERTY EXPORT_NAME detail::c_dependencies)
+
 # utf8proc configuration
 find_package(Utf8Proc REQUIRED)
-if ("${USE_BUNDLED_UTF8PROC}" STREQUAL "TRUE")
+if (${USE_BUNDLED_UTF8PROC})
```

Best to just use the variable name without a dereference:

```suggestion
if (USE_BUNDLED_UTF8PROC)
```

The syntax of `if()` (and `while()`) has some special non-obvious rules ([Refer](https://cmake.org/cmake/help/latest/command/if.html))